---
title: "Fine Mapping With TWAS Results Across Multiple Tissues"
author: "Shuai Li, Xinyu (Brian) Guo"
date: "May 14, 2021"
institute: "Johns Hopkins Bloomberg School of Public Health"
output: 
  beamer_presentation:
    theme: "Ilmenau"
    colortheme: "dolphin"
    fonttheme: "serif"
    slide_level: 2
    toc: true
---

# Background and Methods

## What is TWAS?

- TWAS: transcriptome-wide association study.
- To determine significant trait-expression associations. \footnotemark
- This method increases the power of identifying functionally relevant loci by leveraging expression quantitative trait loci (eQTLs) from external references in relevant tissues. \footnotemark

\footnotetext[1]{Gusev et al. “Integrative approaches for large-scale transcriptome-wide association studies” 2016 Nature Genetics}
\footnotetext[2]{Bhattacharya et al. “A framework for transcriptome-wide association studies in breast cancer in diverse study populations” 2020 Genome Biology}

## TWAS/FUSION Software

\begin{itemize}
\item Functional Summary-based Imputation:

FUSION is a suite of tools for performing a TWAS by predicting functional/molecular phenotypes into GWAS using only summary statistics (usually from GWAS). The goal is to identify associations between a GWAS phenotype and a functional phenotype that was only measured in reference data. \footnotemark[1]
\end{itemize}

\footnotetext[1]{Gusev et al. “Integrative approaches for large-scale transcriptome-wide association studies” 2016 Nature Genetics}

## TWAS/FUSION Software

\begin{itemize}
\item Inputs:
\begin{itemize}
\item GWAS summary statistics
\item Reference panels (i.e. precomputed functional weights (primarily gene expression) from multiple tissues)
\item Regerence LD data
\end{itemize}
\item Outputs: 
\begin{itemize}
\item A data frame with corresponding z and p values for each SNPs.
\end{itemize}
\end{itemize}
\footnotetext[1]{Gusev et al. “Integrative approaches for large-scale transcriptome-wide association studies” 2016 Nature Genetics}

## Bayesian Fine Mapping

\begin{itemize}
\item Why fine-map? 
\begin{itemize}
\item To find causal genes
\item To pinpoint variant
\item To understand genetic architecture
\begin{itemize}
\item Gene enrichment
\item Cross-trait comparison, cross-tissue
\end{itemize}
\end{itemize}
\item Bayesian fine-mapping outputs:
\begin{itemize}
\item PIP: Posterior inclusion probability (the probability that a variant is causal)
\item $95\%$ Credible Sets: Set of variants that contains $\geq95\%$ probability
\end{itemize}
\end{itemize}
\footnotetext[3]{\href{https://www.youtube.com/watch?v=S6vfOr336b0}{\beamergotobutton{Hilary Finucane, Broad Institute}}}

## Bayesian Fine Mapping 
\begin{itemize}
\item Single-causal-variant PIPs:
\begin{eqnarray*}
PIP_{j} &=& P(\text{j causal | data}) \\
        &=& \frac{P(\text{data | j causal})}{\sum_{k} P(\text{data | k causal})}\\
        &=& \frac{P(\text{data | j causal})/P(data | no causal)}{\sum_{k} P(\text{data | k causal})/P(data | no causal)}\\
        &=& \frac{\text{Bayesian Factor}_{j}}{\sum_{k} \text{Bayesian Factor}_{k}}
\end{eqnarray*}
\end{itemize}
\footnotetext[3]{\href{https://www.youtube.com/watch?v=S6vfOr336b0}{\beamergotobutton{Hilary Finucane, Broad Institute}}}

## Bayesian Fine Mapping 
\begin{itemize}
\item $95\%$ Credible Sets (S):
\[
P(\text{causal var is in S})\geq 0.95
\]
\item Under Single-causal-variant assumption:
\[
P(\text{causal var is in S}) = \sum_{j\in S}PIP_{j}
\]
\item To get the most compact credible set, add variant with highest PIPs untill sum to 0.95.
\end{itemize}
\footnotetext[3]{\href{https://www.youtube.com/watch?v=S6vfOr336b0}{\beamergotobutton{Hilary Finucane, Broad Institute}}}

## Bayesian Fine Mapping 
\begin{itemize}
\item Factors affecting Bayesian fine mapping power
\begin{itemize}
\item LD
\item Sample Size
\item Effect size
\end{itemize}
\end{itemize}
\footnotetext[4]{Schaid et al. Nat Rev Genet 2018}

## Bayesian Fine Mapping 
\begin{itemize}
\item Multiple-causal-variant Fine-mapping (two approaches):
\begin{itemize}
\item Divide the whole data into many pieces, and apply single-causal-variant fine-mapping in each piece
\item Jointly model Multiple-causal-variant
\end{itemize}
\end{itemize}
\footnotetext[3]{\href{https://www.youtube.com/watch?v=S6vfOr336b0}{\beamergotobutton{Hilary Finucane, Broad Institute}}}

# Data For Alzheimer's Disease

## Overview
\begin{itemize}
\item Data
\begin{itemize}
\item Gene Expression Matrix: Gene expression level in each tissue
\begin{itemize}
\item Z-values
\item P-values
\end{itemize}
\item Correlation matrix: Correlation of expression in each tissue for each gene
\end{itemize}
\end{itemize}

```{r,echo=F}
load('./ad_TWAS.RData')
load('./cov_matrix.RData')
```

## Gene Expression matrix

```{r}
dim(dat_ad_n[[1]])
dim(dat_ad_n[[2]])
dat_ad_n[[1]][1:5,1:5]
```

## Gene Expression matrix

```{r}
dat_ad_n[[2]][1:5,1:5]

1 - pnorm(dat_ad_n[[1]][3,"Vagina"])
```

## Correlation matrix

```{r}
length(cov_matrix)
names(cov_matrix)[1:10]
```

```{r}
#Obtain correlation
gene = 'EXOC3L2'
cor_matrix <- cov2cor(cov_matrix[[gene]])
```

## Correlation matrix

```{r}
dim(cor_matrix)
round(cor_matrix[15:18,15:18],3)
```

# Analysis (SuSiE)

## SuSiE

```{r,eval=F} 
devtools::install_github("stephenslab/susieR")
library(susieR)
fitted_rss <- susie_rss(z-scores, R, L = 10)
```

\begin{itemize}
\item z-scores: A p-vector of z scores
\item R: p by p correlation matrix
\item L: Maximum number of components model (Credible Sets). 
\end{itemize}



## Implementation

\begin{itemize}
\item run\_susie
\begin{itemize}
\item Pre-process data
\begin{itemize}
\item Drop NA in expression z-scores vector
\item Take out the common tissue information from expression vector and correlation matrix. 
\end{itemize}
\item Fit model: susie\_rss(z-scores, R, L=4)
\begin{itemize}
\item Expression z-score matrix is of length p
\item Correlation matrix is p by p matrix 
\item They contain same tissue information
\item L=4 
\end{itemize} 
\end{itemize}
\end{itemize}

## Main logic

We loop through all genes. For each gene, we implement run\_susie, and take out the significant tissues in Credible Sets (cs), as well as their Posterior inclusion probability (PIP) scores. We stored these information in a csv file.

## Result

```{r,echo=F}
all_res = read.csv('./all_res.csv')
```

```{r}
head(all_res)
```

```{r,echo=F,message=F,warning=F}
library(ggplot2)
library(tidyverse)
library(susieR)

dat_ad <- dat_ad_n[[1]]

run_susie <- function(gene, data, cov_matrix, l = 3){
  dat_gene <- data %>% 
    filter(GENE == gene) %>% 
    select_if(!is.na(.))
  dat_gene <- dat_gene[, -1]
  #check finite
  dat_gene <- dat_gene[, is.finite(as.numeric(dat_gene))] 
  cov_gene <- cov_matrix[[gene]]
  if(ncol(dat_gene) >= ncol(cov_gene)){
    dat_gene = dat_gene %>% select(gsub("\\.", "-", rownames(cov_gene)))
  }
  else{
    cov_gene = cov_gene[gsub("-", "\\.", names(dat_gene)), gsub("-", "\\.", names(dat_gene))]
  }
  dat_input <- as.numeric(dat_gene)
  fitted_rss <- susie_rss(dat_input, cov_gene, L = l)
  vars <- names(dat_gene)
  out <- list(fitted_rss, vars)
  return(out)
}

plot_pip <- function(gene, dat_ad, cov_matrix, all_res){
  #run susie first
  res <- run_susie(gene, dat_ad, cov_matrix, 4)
  fitted_rss = res[[1]]
  var = res[[2]]
  
  #get p-value
  #p.value = (dat_ad.p %>% filter(GENE == gene))[,var]
  
  #get pip value
  pip.value = fitted_rss$pip
  names(pip.value) = var
  
  #get color vector
  color.vec = rep(1,length(pip.value))
  names(color.vec) = names(pip.value)
  
  #this is for p-value
  #names(color.vec) = colnames(p.value)

  cs.group = all_res %>% filter(`GENE`== gene)
  color.vec[cs.group$tissues] = cs.group$cs + 1
  
  #This three lines, I order them by groups. You can comment it to cancel order
  order(color.vec,decreasing = FALSE)
  color.vec = color.vec[order(color.vec,decreasing = FALSE)]
  pip.value = pip.value[names(color.vec)]
  
  ggplot() + 
    geom_point( aes(x = var,y = (as.numeric(pip.value)) ), size=4, alpha=0.7,color=color.vec ) +
    theme_minimal() +
    labs(x="Tissues",y='PIP',title=gene) + 
    theme(plot.title = element_text(hjust=0.5),axis.text.x = element_text(size=10,angle=70, vjust=.8, hjust=0.8))
  
}
```

## Result

```{r}
plot_pip('CD3EAP',dat_ad,cov_matrix,all_res)
```

## Result

```{r}
plot_pip('HLA-DRB1',dat_ad,cov_matrix,all_res)
```

## Result

```{r}
plot_pip('HLA-DQA2',dat_ad,cov_matrix,all_res)
```

## Result

```{r}
plot_pip('HLA-DRA',dat_ad,cov_matrix,all_res)
```

# Reference

## Reference

- Gusev et al. “Integrative approaches for large-scale transcriptome-wide association studies” 2016 Nature Genetics

- Wang, G., Sarkar, A., Carbonetto, P., & Stephens, M. (2020). A simple new approach to variable selection in regression, with application to genetic fine mapping. Journal of the Royal Statistical Society: Series B (Statistical Methodology). https://doi.org/10.1111/rssb.12388

- Schaid, D.J., Chen, W. & Larson, N.B. From genome-wide associations to candidate causal variants by statistical fine-mapping. Nat Rev Genet 19, 491–504 (2018). https://doi.org/10.1038/s41576-018-0016-z

- \href{https://www.youtube.com/watch?v=S6vfOr336b0}{\beamergotobutton{Hilary Finucane, Broad Institute}}



















